src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -833,49 +832,67 @@ static ucs_status_t uct_rc_mlx5_ep_tag_qp_create(uct_rc_mlx5_iface_common_t *ifa
     if (UCT_RC_MLX5_TM_ENABLED(iface)) {
         /* Send queue of this QP will be used by FW for HW RNDV. Driver requires
          * such a QP to be initialized with zero send queue length. */
-        status = uct_rc_iface_qp_create(&iface->super, &ep->tm_qp, &cap, 0);
+        status = uct_rc_iface_qp_create(&iface->super, &ep->tm_qp.verbs.qp, &cap, 0);
         if (status != UCS_OK) {
             return status;
         }
 
-        status = uct_rc_iface_qp_init(&iface->super, ep->tm_qp);
+        status = uct_rc_iface_qp_init(&iface->super, ep->tm_qp.verbs.qp);
         if (status != UCS_OK) {
-            ret = ibv_destroy_qp(ep->tm_qp);
+            ret = ibv_destroy_qp(ep->tm_qp.verbs.qp);
             if (ret) {
                 ucs_warn("ibv_destroy_qp() returned %d: %m", ret);
             }
             return status;
         }
-        uct_rc_iface_add_qp(&iface->super, &ep->super, ep->tm_qp->qp_num);
+        uct_rc_iface_add_qp(&iface->super, &ep->super, ep->tm_qp.verbs.qp->qp_num);
     }
     return UCS_OK;
 }
 
 UCS_CLASS_INIT_FUNC(uct_rc_mlx5_ep_t, const uct_ep_params_t *params)
 {
-    uct_rc_mlx5_iface_common_t *iface;
+    uct_rc_mlx5_iface_common_t *iface = ucs_derived_of(params->iface,
+                                                       uct_rc_mlx5_iface_common_t);
+    struct ibv_qp_cap cap;
     ucs_status_t status;
 
-    iface = ucs_derived_of(params->iface, uct_rc_mlx5_iface_common_t);
+    status = uct_rc_iface_qp_create(&iface->super, &self->qp.verbs.qp,
+                                    &cap, iface->super.config.tx_qp_len);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    self->super.txqp.qp_num = self->qp.verbs.qp->qp_num;
```

Comment:
need to set qp type here. because the verbs part of the qp was created by call to uct_rc_iface_qp_create(), need to set qp type immediately after it returns with success, not later